[DEST-1240] Accept camelCase and snake_case specced video properties

brennan opened the pull request:
https://segment.atlassian.net/browse/DEST-1240
* This PR should not require a CC ticket, as our mobile SDKs are isolated from production services by their very nature.

**What does this PR do?**
- This PR implements a few Nielsen-requested tweaks, in addition to supporting `Video Playback Seek Started`, `Video Playback Buffer Started`, and `Video Playback Buffer Completed`.
- In addition, this PR hard codes the value of `sfcode` to "DCR" per a request from Nielsen.

**Are there breaking changes in this PR?**
- No.

**Any background context you want to provide?**
- n/a

**Is there parity with the server-side/android/iOS integration components (if applicable)?**
- Yes, there will be parity amongst Web, iOS and Android after implementing Nielsen's requested updates. All platforms should be updated in the same week (week of October 14).

**Does this require a new integration setting? If so, please explain how the new setting works**
- n/a

**Links to helpful docs and other external resources**
- n/a

mericsson commented on src/main/java/com/segment/analytics/android/integrations/nielsendcr/NielsenDCRIntegration.java:
```diff
@@ -43,6 +43,81 @@
+  }
+
+  private ValueMap toSegmentSpec(
+      @NonNull ValueMap properties, @NonNull Map<String, String> formatter) {
```
javadoc would be nice on this method

mericsson replied:
Not obvious to me what it is doing.

mericsson commented on src/main/java/com/segment/analytics/android/integrations/nielsendcr/NielsenDCRIntegration.java:
```diff
@@ -166,11 +237,26 @@ private JSONObject mapSpecialKeys(
       contentMetadata.put("segC", segC);
     }
 
+    if (options.containsKey("hasAds") && "true".equals(String.valueOf(options.get("hasAds")))) {
```
I think `options.get("hasAds")` can be `null` which would result in a null pointer exception when `String.valueOf` is applied to it.

mericsson commented on src/main/java/com/segment/analytics/android/integrations/nielsendcr/NielsenDCRIntegration.java:
```diff
@@ -43,6 +43,81 @@
+  private static final Map<String, String> CONTENT_FORMATTER = getContentFormatter();
+
+  private static Map<String, String> getContentFormatter() {
+    Map<String, String> contentFormatter = new LinkedHashMap<>();
```
not a big deal... but curious how you chose `LinkedHashMap` instead of just `HashMap`. I would suggest using HashMap unless you have a particular reason since it raises eyebrows. If you do have a reason, suggest commenting what that reason is.

brennan replied:
Raising eyebrows is def not my goal. 😂  Thanks for calling this out!

CarlosMecha replied:
Do we know how we are going to consume this map? If we are reading key by key (all keys), LinkedHashMap will make sense.

mericsson replied:
nitpick: I think only reason to use LinkedHashMap is if you want to preserve the order of the map. Anyways both will work just fine for what you're doing. :shipit:

mericsson commented on src/main/java/com/segment/analytics/android/integrations/nielsendcr/NielsenDCRIntegration.java:
```diff
@@ -43,6 +43,81 @@
+    return contentFormatter;
+  }
+
+  private static final Map<String, String> AD_FORMATTER = getAdFormatter();
```
nitpick: Could be nice to do `AD_FORMATTER = Collections.unmodifiableMap(getAdFormatter());`

https://stackoverflow.com/a/22636750

brennan commented:
@CarlosMecha @mericsson Addressed all comments, and made some tweaks to our heartbeat timer (the timer will have parity across Web, iOS, Android) in the next version. I'd appreciate a quick re-review when either of you has time. Thanks!
